### DISCUSSION TITLE
Regarding unsafe-no-fsync, a middle ground?

### DISCUSSION THREAD

**Discussion by RaymiiOrg** [etcd-io/etcd#18357] [Q&A]
If the config option `unsafe-no-fsync` is enabled, is there still data saved to disk? For example when the process is stopped?

I'm looking for a middle ground, where my storage is not trashed by writes continuously but for example, every 5 minutes. (Slower harddisks). 

By enabling `unsafe-no-fsync`, is there still data saved but is the operating system in control of when that's synced to disk?

I'm aware of the possibility of data loss and inconsistencies, but those are okay.

**Comment by kien-truong** [comment]
https://github.com/etcd-io/etcd/blob/94077fc7caec762b5ba23cc41c558bc175f3cded/server/storage/wal/wal.go#L800

Yes, the data is still flushed, but no fsync is called, which means it's up to the OS kernel to decide when the data is synced to disk.

**Reply by akalenyu** [reply]
So this configurable is just about WAL? What about the actual commit of the key value data (not the log) to the disk? That keeps getting fsync'd by etcd?

**Reply by kien-truong** [reply]
When you use `unsafe-no-fsync`, the backend (boltdb) is also set to operate in `NoSync` mode, so it'll not issue any fsync when writing data to disk either

https://github.com/etcd-io/etcd/blob/bf63f917059072b5a33ea5fa6b71c5988b1168b7/server/storage/backend/backend.go#L192

**Reply by akalenyu** [reply]
Thank you! So this configurable is both about the WAL log writes & actual data writes not being followed by an fsync?
